spec/features/user_search_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+describe "Filter users by firstname", type: :feature do
+
+     before(:each) do
+        @user1 = User.new(firstname:"Max", lastname:"Mustermann", username:"Maxi", email: "example@example.com")
```

Comment:
wouldn't be `@user1 = FactoryBot.create :user` be sufficient too? Or is there a reason you didn't use the factoryBot?

Reply:
Yes there is, because it creates random names, which is not very good testable

Reply:
but you can test `@user1.firstname ` ?

Reply:
Yes, but what what is a working search_pattern to find this user without enterin his whole name?

Reply:
You could, for example, only use the first 5 characters of the name with `@user1.downcase[0..4]`. IIRC the names are always 26 characters long so there should always be 5 characters in the beginning.

Reply:
But we also test not to have specific text, so how we can exclude that it is randomly on the page?

Reply:
Hm, that's a good point. But it should be sufficient to only set the firstname, shouldn't it? That would already make the line more readable.

Reply:
Hm, that's a good point. Maybe you could add traits to the user factory to specify the attributes needed for your test? Or a nested Factory that only defines these 4 attributes.

Reply:
we are now doing it this way with factorybot and `user[0..4]`